src/Compatibility/Microsoft.DotNet.ApiSymbolExtensions/AttributeDataExtensions.cs

```diff
@@ -33,13 +21,36 @@ public static class AttributeDataExtensions
             attributes.Where(attribute => attribute.IsVisibleOutsideOfAssembly(symbolFilter, excludeWithTypeArgumentsNotVisibleOutsideOfAssembly)).ToImmutableArray();
 
         /// <summary>
-        /// Checks if an <see cref="AttributeData"/> has <see cref="INamedTypeSymbol"/> arguments that point to a <see cref="Type"/> that isn't visible outside of the containing assembly.
+        /// Excludes <see cref="AttributeData"/> based on a passed in filter.
         /// </summary>
-        public static bool HasTypeArgumentsNotVisibleOutsideOfAssembly(this AttributeData attributeData, ISymbolFilter symbolFilter) =>
+        /// <returns></returns>
```

Comment:
I think all this diff can go away if we capture the attribute filter as a composite filter.